std/conv.d

```diff
@@ -3709,12 +3709,30 @@ private S textImpl(S, U...)(U args)
     {
         return null;
     }
+    else static if (U.length == 1)
+    {
+        return args[0].to!S;
+    }
     else
     {
-        auto result = to!S(args[0]);
-        foreach (arg; args[1 .. $])
-            result ~= to!S(arg);
-        return result;
+        import std.array : appender;
+
+        if (__ctfe)
+        {
+            auto result = to!S(args[0]);
+            foreach (arg; args[1 .. $])
+                result ~= to!S(arg);
+            return result;
+        }
+        else
+        {
+            auto app = appender!S();
+            app.reserve(U.length * S.sizeof);
```

Comment:
Why `S.sizeof`? As far as I see, it's just an arbitrary number, determined by rule of thumb, and using `S.sizeof` for this suggests a relation where there is none.